test/Parse/collection_conditionals_error.swift

```diff
@@ -0,0 +1,46 @@
+// RUN: %target-typecheck-verify-swift
+
+[
+  #if FLAG
+  "foo",
+  #else
+  "foo": 12,
+  #endif
+]
+// expected-error@-3{{expected ',' separator}}
+// expected-error@-4{{expected expression in container literal}}
+// expected-warning@-9{{expression of type '[String]' is unused}}
```

Comment:
Could you use trailing `expected-error` style? i.e.
```swift
[ // expected-warning {{expression of type '[String]' is unused}}
   #if FLAG 
   "foo",
   #else
   "foo": 12, // expected-error {{expected ',' separator}}
              // expected-error@-1 {{expected expression in container literal}}
   #endif
 ]
```
This is easier to understand.

Reply:
Much better!